source/src/BNFC/Backend/Java.hs

```diff
@@ -97,138 +121,495 @@ makeJava options@Options{..} cf =
+mkCFtoLexer :: CF2LexerFunction -> MakeFileDetails -> CFToLexer
+mkCFtoLexer fu mf = CF2Lex {
+    cf2lex = fu,
+    makelexerdetails = mf
+}
+
+-- | Instances of cf-lexergen bridges
+cf2JLex, cf2JFlex:: CFToLexer
+cf2JLex     = mkCFtoLexer BNFC.Backend.Java.CFtoJLex15.cf2jlex' jlexmakedetails
+cf2JFlex    = mkCFtoLexer BNFC.Backend.Java.CFtoJLex15.cf2jflex' jflexmakedetails
+cf2AntlrLex' :: String -> CFToLexer
+cf2AntlrLex' l = mkCFtoLexer BNFC.Backend.Java.CFtoAntlr4Lexer.cf2AntlrLex $ antlrmakedetails $ l++"Lexer" -- TODO
+
+
+-- | CF -> PARSER GENERATION TOOL BRIDGE
+-- | type of function translating the CF to an appropriate parser generation tool.
+type CF2ParserFunction = String -> String -> CF -> SymEnv -> String
+-- | Chooses the translation from CF to the parser
+data CFToParser = CF2Parse {
+    cf2parse :: CF2ParserFunction,
+    makeparserdetails :: MakeFileDetails
+}
+-- |  Shorthand for positional constructor
+mkCFtoParser :: CF2ParserFunction -> MakeFileDetails -> CFToParser
+mkCFtoParser fu mf = CF2Parse {
+    cf2parse = fu,
+    makeparserdetails = mf
+}
+-- | Instances of cf-parsergen bridges
+cf2cup :: CFToParser
+cf2cup = mkCFtoParser BNFC.Backend.Java.CFtoCup15.cf2Cup cupmakedetails
+cf2AntlrParse' :: String -> CFToParser
+cf2AntlrParse' l = mkCFtoParser BNFC.Backend.Java.CFtoAntlr4Parser.cf2AntlrParse $ antlrmakedetails $ l++"Parser"
+
+
+-- | shorthand for Makefile command running javac or java
+runJavac , runJava:: String -> String
+runJava s = mkRunProgram "JAVA" s
+runJavac s = mkRunProgram "JAVAC" s
+
+-- | function returning a string executing a program contained in a variable j on input s
+mkRunProgram :: String -> String -> String
+mkRunProgram j s = (Makefile.refVar j) +++ (Makefile.refVar (j +-+ "FLAGS")) +++ s
+
+
+type OutputDirectory = String
+
+-- | MAKEFILE DETAILS FROM RUNNING THE PARSER-LEXER GENERATION TOOLS
+-- | executable : the string that executes the generation tool
+-- | flags : a string containing flags to pass to the tool
+-- | filename : the input file to the tool
+-- | toolname : name of the tool
+-- | toolversion : tool version
+-- | supportsEntryPoints : true if the tool is a parser and supports entry points, false otherwise
+-- | results : list of names (without extension!) of files resulting
+--             from the application of the tool which are relevant to
+--             a make rule
+-- | moveresults : if true, the files are moved to the base directory, otherwise they are left where they are
+data MakeFileDetails = MakeDetails {
+    executable::String,
+    flags::(OutputDirectory -> String),
+    filename::String,
+    fileextension::String,
+    toolname::String,
+    toolversion::String,
+    supportsEntryPoints::Bool,
+    results :: [String],
+    moveresults :: Bool
+}
+
+-- Positional constructor.
+mkMakeFileDetails :: String -> (OutputDirectory->String) -> String -> String -> String -> String -> Bool -> [String] -> Bool -> MakeFileDetails
```

Comment:
Mm now I understand why I did so for some types. It was for sparing me to write the name of the field to initialize.
Nevertheless I will follow your advice, the code will anyway be more readable and less prone to have bugs introduced.
it is hard to say which is which when you use the positional constructors.